pkg/controller/operators/catalog/step.go

```diff
@@ -0,0 +1,160 @@
+package catalog
+
+import (
+	"fmt"
+	"k8s.io/client-go/dynamic"
+	"k8s.io/client-go/tools/cache"
+
+	"github.com/operator-framework/operator-lifecycle-manager/pkg/api/apis/operators/v1alpha1"
+	crdlib "github.com/operator-framework/operator-lifecycle-manager/pkg/lib/crd"
+	index "github.com/operator-framework/operator-lifecycle-manager/pkg/lib/index"
+	"github.com/operator-framework/operator-lifecycle-manager/pkg/lib/operatorclient"
+	errorwrap "github.com/pkg/errors"
+	logger "github.com/sirupsen/logrus"
+	apiextensionsv1 "k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1"
+	"k8s.io/apiextensions-apiserver/pkg/client/clientset/clientset"
+	k8serrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// Stepper manages cluster interactions based on the step.
+type Stepper interface {
+	Status() (v1alpha1.StepStatus, error)
```

Comment:
The naming here is a little confusing. Usually I would expect <Verb>er to <Verb>.

Reply:
Agreed